### DISCUSSION TITLE
Odd generic issue only when trait is used

### DISCUSSION THREAD

**Discussion by KorvinSzanto** [vimeo/psalm#10712] [Q&A]
Compare the following two sandboxes, the only change is the few lines at the very bottom:

- Without use: https://psalm.dev/r/95ef462230 No issues
- With use: https://psalm.dev/r/2d87994b3d 3 errors

What's causing the three errors in the second sandbox?

---
I'm seeing separate issues when trying to use my generic trait with imported types. If the type is imported, the keys don't seem to resolve. Compare the following two sandboxes, one uses an imported type while the other uses a local type.

- Imported type: https://psalm.dev/r/25308433cd Additional DocblockTypeContradiction error
- Local type: https://psalm.dev/r/2c5e9345a0 Only the 3 errors from the first issue.

Is this a bug with imported types, or am I misunderstanding how imported types are meant to be used?

**Comment by weirdan** [comment]
> Without use: [psalm.dev/r/95ef462230](https://psalm.dev/r/95ef462230) No issues

This part is simple: traits are not checked at all, unless used. Simplified example: https://psalm.dev/r/a55aa42a2d

> What's causing the three errors in the second sandbox?

Bugs, I suppose.

> Is this a bug with imported types, or am I misunderstanding how imported types are meant to be used?

More likely with index types.

**Reply by KorvinSzanto** [reply]
That helps answer why zero errors but doesn't really get me closer to writing this without errors.
